hist/hist/src/TMultiGraph.cxx

```diff
@@ -1602,21 +1604,22 @@ void TMultiGraph::SavePrimitive(std::ostream &out, Option_t *option)
    while (auto g = iter())
       g->SavePrimitive(out, TString::Format("multigraph%s", iter.GetOption()).Data());
 
+   if (fHistogram) {
+      TString hname = fHistogram->GetName();
+      fHistogram->SetName(TString::Format("MGraph_histogram%d", ++frameNumber).Data());
+      fHistogram->SavePrimitive(out, "nodraw");
+      out << "   " << fHistogram->GetName() << "->GetXaxis()->SetLimits(" << fHistogram->GetXaxis()->GetXmin() << ", " << fHistogram->GetXaxis()->GetXmax() << ");\n";
+      out << "   multigraph->SetHistogram(" << fHistogram->GetName() << ");\n";
+      out << "   \n";
+      fHistogram->SetName(hname.Data());
+   }
+
    const char *l = strstr(option, "th2poly");
    if (l)
       out << "   " << l + 7 << "->AddBin(multigraph);\n";
    else
       SavePrimitiveDraw(out, "multigraph", option);
 
-   TAxis *xaxis = GetXaxis();
-   TAxis *yaxis = GetYaxis();
-
-   if (xaxis) {
```

Comment:
This removal is not called out in the commit log/PR header.  Is it related/replaced-by storing the histogram?

Reply:
It is point 4 - "For TMultiGraph store histogram like for TGraph". 
Before TMultiGraph tried to store axes attributes - making canvas update when storing multigraph in the macro.
And this has different side effects.